index.js

```diff
@@ -221,7 +221,8 @@ async function fastifyStatic (fastify, opts) {
       }
     }
 
-    const stream = send(request.raw, pathnameForSend, options)
+    // `send(..., path, ...)` will URI-decode path so we pass an encoded path here
+    const stream = send(request.raw, encodeURI(pathnameForSend), options)
```

Comment:
your comment: we uri decode path in send? How about apply a change in [@fastify/send ](https://github.com/fastify/send), like pass decodeUri option in options, by default true for backwards compat and we set it here to false and then we dont need to encodeURI?

Reply:
From the docs of `@fastify/send` is seems the path passed is intended to be URL-encoded:

> `path` is a urlencoded path to send (urlencoded, not the actual file-system path)

https://github.com/fastify/send?tab=readme-ov-file#api

While I agree that this seems like a little extra overhead it looks like the `@fastify/send` API was not being used correctly.

Reply:
Maybe send could be refactored to do what Uzlopak suggests